src/components/fields/relationship/one2many.js

```diff
@@ -110,33 +107,26 @@ defineComponent('furet-ui-form-field-one2many', {
       />
     </furet-ui-form-field-common-tooltip>
   `,
-  extend: ['furet-ui-form-field-common'],
+  extend: ["furet-ui-form-field-common"],
   prototype: {
     methods: {
       addState(actions, state) {
-        const pks = pk2string(actions.pks)
-        const newvalue = _.map(this.value, value => {
-          if (pk2string(value) === pks) {
-            value.__x2m_state = state;
-          }
-          return value;
-        })
-        return newvalue;
+        return [Object.assign({}, actions.pks, { __x2m_state: state})];
       },
-      o2m_add (actions) {
-        const newvalue = _.map(this.value, value => value)
-        if (!_.find(newvalue, value => value.uuid === actions.uuid)) {
-          newvalue.push({__x2m_state: 'ADDED', uuid: actions.uuid})
-        }
-        this.updateValue(newvalue, actions.changes)
+      o2m_add(actions) {
+        const newvalue = [];
+        newvalue.push({ __x2m_state: "ADDED", uuid: actions.uuid });
+        this.updateValue(newvalue, actions.changes);
       },
-      o2m_update (actions) {
-        this.updateValue(this.addState(actions, 'UPDATED'), actions.changes)
+      o2m_update(actions) {
+        this.updateValue(this.addState(actions, "UPDATED"), actions.changes);
       },
-      o2m_delete (actions) {
-        this.updateValue(this.addState(actions, 'DELETED'))
-      },
-    },
+      o2m_delete(actions) {
+        this.updateValue(this.addState(actions, "DELETED"));
+        //TODO: figure out why this commit is not taken in consideration
+        this.$store.commit("DELETE_DATA", actions);
```

Comment:
I couldn't figure out why this was not updating the state of the store properly.

Reply:
In fact I understood...
The commit is properly done, the matter is that when the user come back to the o2m list (delete button is on the form resource) a call to the server is made but we haven't ask the server to save that change yet.

I think we shouldn't call the server until the main save.

Or consider this component independent and save changes in x2m in a different transaction.... which is not what we wants !

Reply:
Yes it is the reason why we saved the list of primary keys